src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -602,4 +606,68 @@ String getAppPath(String appName) {
     List<String> getTargetSpecificObjectFiles() throws IOException {
         return Collections.emptyList();
     }
+
+    /**
+     * For every jar in the classpath, checks for native libraries (*.a)
+     * and if found, extracts them to a folder, for later link
+     *
+     * @param classPath The classpath of the project
+     * @throws IOException
+     */
+    private void extractNativeLibs(String classPath) throws IOException {
+        Path libPath = paths.getGvmPath().resolve(Constants.LIB_PATH);
+        Logger.logDebug("Extracting native libs to: " + libPath);
+        String[] split = classPath.split(File.pathSeparator);
+        List<String> jars = Stream.of(split)
+                .filter(s -> s.endsWith(".jar") && !s.contains("javafx-"))
+                .collect(Collectors.toList());
+        for (String jar : jars) {
+            FileOps.extractFilesFromJar(".a", Path.of(jar), libPath, getTargetSpecificNativeLibsFilter());
+        }
+    }
+
+    /**
+     * A filter can be used to verify if the native library matches certain
+     * criteria, like being available for a given architecture
+     *
+     * @return a predicate, default is null (no filter applied)
+     */
+    Predicate<Path> getTargetSpecificNativeLibsFilter() {
+        return null;
```

Comment:
would a default that does nothing be more appropriate?

Reply:
This is the default, a null predicator that let the file pass the filter. We could return `p -> true`, but with null we don't need to evaluate it.